samples/openapi3/client/petstore/python-experimental/tests/test_http_signature.py

```diff
@@ -0,0 +1,370 @@
+            signed_headers=[
+                signing.HEADER_REQUEST_TARGET,
+                signing.HEADER_CREATED,
+                signing.HEADER_HOST,
+                signing.HEADER_DATE,
+                signing.HEADER_DIGEST,
+                'Content-Type'
+            ]
+        )
+        config = Configuration(host=HOST, signing_info=signing_cfg)
+        # Set the OAuth2 acces_token to None. Here we are interested in testing
+        # the HTTP signature scheme.
+        config.access_token = None
+
+        api_client = petstore_api.ApiClient(config)
+        pet_api = petstore_api.PetApi(api_client)
+
+        mock_pool = MockPoolManager(self)
+        api_client.rest_client.pool_manager = mock_pool
+
+        mock_pool.expect_request('POST', 'http://petstore.swagger.io/v2/pet',
+                                 body=json.dumps(api_client.sanitize_for_serialization(self.pet)),
+                                 headers={'Content-Type': 'application/json',
+                                          'Authorization': 'Signature keyId="my-key-id",algorithm="hs2019",created=[0-9]+,'
+                                                'headers="\(request-target\) \(created\) host date digest content-type",'
+                                                'signature="[a-zA-Z0-9+/]+="',
+                                          'User-Agent': 'OpenAPI-Generator/1.0.0/python'},
+                                 preload_content=True, timeout=None)
+
+        pet_api.add_pet(self.pet)
+
+    def test_valid_http_signature_with_defaults(self):
+        signing_cfg = signing.HttpSigningConfiguration(
+            key_id="my-key-id",
+            private_key_path=self.rsa4096_key_path,
+            signing_scheme=signing.SCHEME_HS2019,
+        )
+        config = Configuration(host=HOST, signing_info=signing_cfg)
+        # Set the OAuth2 acces_token to None. Here we are interested in testing
+        # the HTTP signature scheme.
+        config.access_token = None
+
+        api_client = petstore_api.ApiClient(config)
+        pet_api = petstore_api.PetApi(api_client)
+
+        mock_pool = MockPoolManager(self)
+        api_client.rest_client.pool_manager = mock_pool
+
+        mock_pool.expect_request('POST', 'http://petstore.swagger.io/v2/pet',
+                                 body=json.dumps(api_client.sanitize_for_serialization(self.pet)),
+                                 headers={'Content-Type': 'application/json',
+                                          'Authorization': 'Signature keyId="my-key-id",algorithm="hs2019",created=[0-9]+,'
+                                                'headers="\(created\)",'
+                                                'signature="[a-zA-Z0-9+/]+="',
+                                          'User-Agent': 'OpenAPI-Generator/1.0.0/python'},
+                                 preload_content=True, timeout=None)
+
+        pet_api.add_pet(self.pet)
```

Comment:
Can you decode the request using the key to show that you ket back the unencrypted data to verify that this is working?

Reply:
I have added unit tests to verify the signature with the public key.